lib/puppetserver/ca/action/prune.rb

```diff
@@ -69,18 +76,26 @@ def prune_CRLs(crl_list)
               if existed_serial_number.add?(revoked.serial)
                 false
               else
+                number_of_removed_duplicates += 1
                 @logger.debug("Removing duplicate of #{revoked.serial}, " \
                   "revoked on #{revoked.time}\n") if @logger.debug?
                 true
               end
             end
             crl.revoked=(revoked_list)
           end
+
+          return number_of_removed_duplicates
         end
 
         def update_pruned_CRL(crl_list, pkey)
           crl_list.each do |crl|
-            crl.version=(crl.version + 1)
+            number_ext, other_ext = crl.extensions.partition{ |ext| ext.oid == "crlNumber" }
+            number_ext.each do |crl_number|
+              updated_crl_number = OpenSSL::BN.new(crl_number.value) + OpenSSL::BN.new(1)
```

Comment:
Why do we need `OpenSSL::BN`s now?

Reply:
It was one those thing that the doc wasn't being really clear about what `OpenSSL::ASN1::Integer` take for parameter. I found out that `OpenSSL::ASN1::Integer()` take a `OpenSSL::BN` object so I figured I stay as close as I can to the doc and used `OpenSSL::BN`.  I was following how we generated the `crlNumber` in the utils file and may have went over the edge 😅

Reply:
yeah, the docs are definitely confusing. I think this is fine, but based on https://ruby-doc.org/stdlib-2.7.0/libdoc/openssl/rdoc/OpenSSL/ASN1.html it looks like just `OpenSSL::ASN1::Integer.new(number)` would work.

Reply:
Oh yeah. The main reason create the `updated_crl_number` variable is because of how the `extension.value` function return a string.  Luckily `OpenSSL::BN.new` can accept a string so that why I opted to use `OpenSSL::BN` so that it is uniform everywhere and doesn't cause lossage (is that a word?) of bytes.

Reply:
ahh, gotcha! that seems like a good approach to me, thanks for explaining!